uk.co.compucorp.civicrm.hrleaveandabsences/js/src/leave-absences/shared/controllers/sub-controllers/request-modal-details-leave.controller.js

```diff
@@ -26,6 +27,52 @@ define([
     detailsController.resetUIInputsExtended = resetUITimesDayTypesAndDeductions;
     detailsController.setDaysSelectionModeExtended = setDaysSelectionModeExtended;
 
+    /**
+     * Adjusts the time for a work patterns day depending on the day type.
+     * It allows values accoring to the logic below:
+     * - "start" day time 00:00 to 23:30
+     * - "end" day time 00:15 to 23:45
+     *
+     * @param  {String} time
+     * @param  {String} timeType from|to
+     * @return {String} adjusted time
+     */
+    function adjustWorkPatternTime (time, timeType) {
+      var timeBaseInMinutes = sharedSettings.timeBaseInMinutes;
+      var momentTime = moment(time, timeFormat);
+      var minTime = moment('00:00', timeFormat);
+      var maxTime = moment('23:45', timeFormat);
+
+      (timeType === 'to') && minTime.add(timeBaseInMinutes, 'minutes');
+      (timeType === 'from') && maxTime.subtract(timeBaseInMinutes, 'minutes');
+
+      if (momentTime.isBefore(minTime)) {
+        momentTime = minTime;
+      } else if (momentTime.isAfter(maxTime)) {
+        momentTime = maxTime;
+      }
+
+      return momentTime.format(timeFormat);
+    }
+
+    /**
+     * Rounds work day times according to the base, if times exist.
+     *
+     * @param {Object} workDay object resolved by leaveRequest.getWorkDayForDate()
+     */
+    function adjustWorkPatternTimes (workDay) {
+      ['from', 'to'].forEach(function (timeType) {
+        var timeKey = ['time_' + timeType];
+
+        if (!workDay[timeKey]) {
+          return;
+        }
+
+        workDay[timeKey] = adjustWorkPatternTime(workDay[timeKey], timeType);
+        workDay[timeKey] = roundWorkPatternTime(workDay[timeKey], timeType);
+      });
```

Comment:
For me `adjustWorkPatternTime` is confusing. I can't tell at a glance what it does. What's the difference between `adjustWorkPatternTime` and `roundWorkPatternTime` ?

Maybe renaming the function to something like `limitWorkPatternTimeToMinMaxRange`, or a more specific name. The problem for me is that "adjust" could be anything.

Reply:
Done.